roles/KubevirtCommonTemplatesBundle/tasks/main.yml

```diff
@@ -5,21 +5,64 @@
     state: present
     namespace: "{{ meta.namespace }}"
     definition: "{{ item | from_yaml }}"
+    apply: yes
   with_items: "{{ lookup('file', 'common-templates-'+ version +'.yaml').split('\n---\n') | select('search', '(^|\n)[^#]') | list }}"
   register: ct_status
 
+# Get all templates
+- name: Fetching all templates
+  set_fact:
+    templates: "{{ lookup('k8s', api_version=ct_status.results[0].result.apiVersion, kind='template') }}"
+
+- name: Fetch old cr
+  block:
+  - set_fact:
+      old_cr: "{{ lookup('k8s', api_version='kubevirt.io/v1', kind='KubevirtCommonTemplatesBundle') }}"
+  - set_fact:
+      old_cr_exists: true
+  rescue:
+  - set_fact:
+      old_cr_exists: false
+
+- name: Filter for templates owned by the old cr
+  set_fact:
+    old_cr_templates: "{{ (old_cr_templates | default([])) + ([item] if (item.metadata.ownerReferences is defined and item.metadata.ownerReferences[0].uid==old_cr.metadata.uid) else []) }}"
```

Comment:
I do not see the filtering here.. It is also pretty unreadable. I suggest writing a python filter and then using for example:

`{{ old_cr_templates |k8s_owned_by(old_cr) }}`

A example of how to do that:
- create `filter_plugins` dir in the role (as sibling to tasks dir)
- in there create a file `k8s.py`

And in the file write something like:

```
def k8s_owned_by(objects, owner):
    return [o for o in objects if ....]

class FilterModule(object):
    '''Extra filters...'''
    def filters(self):
        return {
            'k8s_owned_by': k8s_owned_by
        }
```

Reply:
This task iterates over all templates fetched by the `Fetching all templates` task, and appends all templates that have the `old_cr` as their owner to `old_cr_templates`